fv3core/stencils/dyn_core.py

```diff
@@ -179,381 +190,413 @@ def dyncore_temporaries(shape):
-        # else:
-        #    divg2 = utils.make_storage_from_shape(delz.shape, grid.compute_origin())
-
-        if not hydrostatic:
-            updatedzd.compute(
+        if not self.namelist.hydrostatic:
+            # k1k = akap / (1.0 - akap)
+            self._dp_ref_compute(
+                state.ak,
+                state.bk,
+                state.phis,
                 state.dp_ref,
                 state.zs,
-                state.zh,
-                state.crx,
-                state.cry,
-                state.xfx,
-                state.yfx,
-                state.wsd,
-                dt,
+                rgrav,
             )
+            # After writing, make 'dp_ref' a K-field and 'zs' an IJ-field
+            state.dp_ref = utils.make_storage_data(
+                state.dp_ref[0, 0, :], (shape[2],), (0,)
+            )
+            state.zs = utils.make_storage_data(state.zs[:, :, 0], shape[0:2], (0, 0))
+
+        # "acoustic" loop
+        # called this because its timestep is usually limited by horizontal sound-wave
+        # processes. Note this is often not the limiting factor near the poles, where
+        # the speed of the polar night jets can exceed two-thirds of the speed of sound.
+        for it in range(n_split):
+            # the Lagrangian dynamics have two parts. First we advance the C-grid winds
+            # by half a time step (c_sw). Then the C-grid winds are used to define
+            # advective fluxes to advance the D-grid prognostic fields a full time step
+            # (the rest of the routines).
+            #
+            # Along-surface flux terms (mass, heat, vertical momentum, vorticity,
+            # kinetic energy gradient terms) are evaluated forward-in-time.
+            #
+            # The pressure gradient force and elastic terms are then evaluated
+            # backwards-in-time, to improve stability.
+            remap_step = False
+            if self.namelist.breed_vortex_inline or (it == n_split - 1):
+                remap_step = True
+            if not self.namelist.hydrostatic:
+                if self.do_halo_exchange:
+                    reqs["w_quantity"] = self.comm.start_halo_update(
+                        state.w_quantity, n_points=self.grid.halo
+                    )
+                if it == 0:
+                    self._set_gz(
+                        state.zs,
+                        state.delz,
+                        state.gz,
+                    )
+                    if self.do_halo_exchange:
+                        reqs["gz_quantity"] = self.comm.start_halo_update(
+                            state.gz_quantity, n_points=self.grid.halo
+                        )
+            if it == 0:
+                if self.do_halo_exchange:
+                    reqs["delp_quantity"].wait()
+                    reqs["pt_quantity"].wait()
+
+            if it == n_split - 1 and end_step:
+                if self.namelist.use_old_omega:  # apparently True
```

Comment:
Not from your PR so this is optional, but do you know what "apparently True" means? Could you expand the comment or delete it?

Reply:
deleted